tests/legacy-cli/e2e/tests/misc/target-default-configuration.ts

```diff
@@ -0,0 +1,35 @@
+import { expectFileToExist } from '../../utils/fs';
+import { ng } from '../../utils/process';
+import { updateJsonFile } from '../../utils/project';
+import { expectToFail } from '../../utils/utils';
+
+export default async function () {
+  await updateJsonFile('angular.json', workspace => {
+    const build = workspace.projects['test-project'].architect.build;
+    build.defaultConfiguration = undefined;
+    build.options = {
+      ...build.options,
+      optimization: false,
+      buildOptimizer: false,
+      outputHashing: 'none',
+      sourceMap: true,
+    };
+  });
+
+  await ng('build');
+  await expectFileToExist('dist/test-project/main.js');
+  await expectFileToExist('dist/test-project/main.js.map');
+
+  // Add new configuration and set "defaultConfiguration"
+  await updateJsonFile('angular.json', workspace => {
+    const build = workspace.projects['test-project'].architect.build;
+    build.defaultConfiguration = 'defaultConfiguration';
```

Comment:
Can you use a different name for the configuration (`my-default`, `foo`, etc.) ?  This may be looked at as an example and the configuration having the same name as the property may cause confusion.